demos/common/cpp/models/src/associative_embedding_decoder.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+// Copyright (C) 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include <utils/common.hpp>
+#include "utils/kuhn_munkres.hpp"
+
+#include "models/associative_embedding_decoder.h"
+
+
+void findPeaks(const std::vector<cv::Mat>& nmsHeatMaps,
+               const std::vector<cv::Mat>& aembdsMaps,
+               std::vector<std::vector<Peak>>& allPeaks,
+               int jointId, const int maxNumPeople) {
+
+    const cv::Mat& nmsHeatMap = nmsHeatMaps[jointId];
+    const float* heatMapData = nmsHeatMap.ptr<float>();
+    cv::Size outputSize = nmsHeatMap.size();
+
+    std::vector<int> indices(outputSize.area());
+    std::iota(std::begin(indices), std::end(indices), 0);
+    std::partial_sort(std::begin(indices), std::begin(indices) + maxNumPeople, std::end(indices),
+                      [&heatMapData](int l, int r) { return heatMapData[l] > heatMapData[r]; });
```

Comment:
```suggestion
                      [heatMapData](int l, int r) { return heatMapData[l] > heatMapData[r]; });
```
`heatMapData` is a ptr. You can copy it.